plugins/module_utils/oracle/oci_utils.py

```diff
@@ -1026,10 +1026,10 @@ def check_if_user_value_matches_resources_attr(
             return
 
         if (
-            resources_value_for_attr is None
-            and len(user_provided_value_for_attr) >= 0
-            or user_provided_value_for_attr is None
-            and len(resources_value_for_attr) >= 0
+            (resources_value_for_attr is None
+             and user_provided_value_for_attr is not None) or
+            (user_provided_value_for_attr is None
+             and resources_value_for_attr is not None)
```

Comment:
The same is true here:
```suggestion
            resources_value_for_attr is None or user_provided_value_for_attr is None
```